aws/resource_aws_appsync_datasource.go

```diff
@@ -47,6 +47,7 @@ func resourceAwsAppsyncDatasource() *schema.Resource {
 					appsync.DataSourceTypeAmazonElasticsearch,
 					appsync.DataSourceTypeHttp,
 					appsync.DataSourceTypeNone,
+					appsync.DataSourceTypeRelationalDatabase,
```

Comment:
As the author of #16480, I note this could be replaced with `ValidateFunc: validation.StringInSlice(appsync.DataSourceType_Values(), false),`